### ISSUE TITLE
Style/TrivialAccessors autocorrects def self.foo? into invalid attr_reader :foo?

### ISSUE THREAD

**Issue by danlo** [rubocop/rubocop#15428]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/TrivialAccessors

### Steps to reproduce the problem

Created a repo here with the full steps and recommendation by Fable 5.

https://github.com/danlo/trivial_accessor_bug/

### Expected behavior

See above

### Actual behavior

See above

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
main$ bundle exec rubocop -V
1.88.1 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.4.9) [x86_64-linux]
```

**Comment by danlo** [comment]
Thank you!
